test/import/FSHImporter.Mapping.test.ts

```diff
@@ -0,0 +1,203 @@
+import { importSingleText } from '../testhelpers/importSingleText';
+import { assertMappingRule } from '../testhelpers/asserts';
+import { loggerSpy } from '../testhelpers/loggerSpy';
+import { FshCode } from '../../src/fshtypes';
+
+describe('FSHImporter', () => {
+  beforeAll(() => {
+    loggerSpy.reset();
+  });
+
+  describe('Mapping', () => {
+    it('should parse the simplest possible Mapping', () => {
+      const input = `
+        Mapping: MyMapping
+        `;
+      const result = importSingleText(input, 'Mapping.fsh');
+      expect(result.mappings.size).toBe(1);
+      const mapping = result.mappings.get('MyMapping');
+      expect(mapping.name).toBe('MyMapping');
+      expect(mapping.id).toBe('MyMapping');
+      expect(mapping.sourceInfo.location).toEqual({
+        startLine: 2,
+        startColumn: 9,
+        endLine: 2,
+        endColumn: 26
+      });
+      expect(mapping.sourceInfo.file).toBe('Mapping.fsh');
+    });
+
+    it('should parse a Mapping with additional metadata properties', () => {
+      const input = `
+        Mapping: MyMapping
+        Id: my-map
+        Source: Patient
+        Target: "http://some.com/mappedTo"
+        Description: "This is a description"
+        `;
+      const result = importSingleText(input, 'Mapping.fsh');
+      expect(result.mappings.size).toBe(1);
+      const mapping = result.mappings.get('MyMapping');
+      expect(mapping.name).toBe('MyMapping');
+      expect(mapping.id).toBe('my-map');
+      expect(mapping.source).toBe('Patient');
+      expect(mapping.target).toBe('http://some.com/mappedTo');
+      expect(mapping.description).toBe('This is a description');
+      expect(mapping.sourceInfo.location).toEqual({
+        startLine: 2,
+        startColumn: 9,
+        endLine: 6,
+        endColumn: 44
+      });
+      expect(mapping.sourceInfo.file).toBe('Mapping.fsh');
+    });
+
+    it('should only apply each metadata attribute the first time it is declared', () => {
+      const input = `
+        Mapping: MyMapping
+        Id: my-map
+        Source: Patient
+        Target: "http://some.com/mappedTo"
+        Description: "This is a description"
+        Id: my-map-2
+        Source: Patient2
+        Target: "http://some.com/mappedTo2"
+        Description: "This is a description 2"
+        `;
+      const result = importSingleText(input, 'Mapping.fsh');
+      expect(result.mappings.size).toBe(1);
+      const mapping = result.mappings.get('MyMapping');
+      expect(mapping.name).toBe('MyMapping');
+      expect(mapping.id).toBe('my-map');
+      expect(mapping.source).toBe('Patient');
+      expect(mapping.target).toBe('http://some.com/mappedTo');
+      expect(mapping.description).toBe('This is a description');
+      expect(loggerSpy.getLastMessage('error')).toMatch(
+        /This is a description.*File: Mapping\.fsh.*Line: 10\D*/s
+      );
+    });
+
+    it('should only apply each metadata attribute the first time it is declared', () => {
```

Comment:
What is the difference between this test and the one above besides checking the error that is logged in the test above? Do we need this test?

Reply:
Nope, good point!